pyccl/tracers.py

```diff
@@ -264,6 +296,133 @@ def get_bessel_derivative(self):
 
         return np.array([t.der_bessel for t in self._trc])
 
+    def _MG_add_tracer(self, cosmo, kernel, z_b, der_bessel=0, der_angles=0,
+                       bias_transfer_a=None, bias_transfer_k=None):
+        """ function to set mg_transfer in the right format and add MG tracers
+            for different cases including different cases and biases like
+            intrinsic alignements (IA) when present
+        """
+        # Getting MG transfer function and building a k-array
+        mg_transfer, k = self._get_MG_transfer_function(cosmo, z_b)
+
+        # case with no astro biases
+        if ((bias_transfer_a is None) and (bias_transfer_k is None)):
+            if (isinstance(k, (list, np.ndarray))):
```

Comment:
OK, now we know that `k` is always an array, right? In which case, do you need these `if` statements?
It's good to avoid unnecessary `if`s because otherwise coveralls will complain that you're not testing all possible if-branches of the code.

Reply:
For the further above comment: Very good question and I agree that completes the definition, thanks! I added a clear statement for f1 and f2.

I also agree for the unnecessary ifs -- all removed!